libdnf/hy-package.cpp

```diff
@@ -1015,7 +1016,9 @@ dnf_package_get_advisories(DnfPackage *pkg, int cmp_type)
             (cmp < 0 && (cmp_type & HY_LT)) ||
             (cmp == 0 && (cmp_type & HY_EQ))) {
             advisory = dnf_advisory_new(sack, di.solvid);
-            g_ptr_array_add(advisorylist, advisory);
+            if (libdnf::isAdvisoryApplicable(*advisory, sack)) {
```

Comment:
This introduces a memory leak.  The following patch fixes it:
```
--- a/libdnf/hy-package.cpp
+++ b/libdnf/hy-package.cpp
@@ -1019,6 +1019,9 @@ dnf_package_get_advisories(DnfPackage *pkg, int cmp_type)
             if (libdnf::isAdvisoryApplicable(*advisory, sack)) {
                 g_ptr_array_add(advisorylist, advisory);
             }
+            else {
+                delete advisory;
+            }
             dataiterator_skip_solvable(&di);
         }
     }
```

Reply:
@kdudka  Thanks a lot.